magicctapipe/scripts/lst1_magic/lst1_magic_event_coincidence.py

```diff
@@ -225,20 +256,80 @@ def event_coincidence(input_file_lst, input_dir_magic, output_dir, config):
 
     for tel_id in tel_ids:
-
         tel_name = TEL_NAMES[tel_id]
         df_magic = event_data_magic.query(f"tel_id == {tel_id}").copy()
 
         # Arrange the MAGIC timestamps as same as the LST timestamps
         seconds = np.array([Decimal(str(time)) for time in df_magic["time_sec"]])
         nseconds = np.array([Decimal(str(time)) for time in df_magic["time_nanosec"]])
-
         timestamps_magic = seconds * SEC2NSEC + nseconds
+
+        if input_dir_toff is not None:
+            files_toff = glob.glob(
+                input_dir_toff + "/*M" + str(tel_id - 1) + "*_detail.npy"
+            )
+            # files_toff = glob.glob(input_dir_toff)
+            print(files_toff)
+            i = 0
+            for file_toff in files_toff:
+                file_npy = np.load(file_toff)
+                if i == 0:
+                    file_npy_ = file_npy
+                if i != 0:
+                    file_npy_ = np.hstack([file_npy, file_npy_])
+                i += 1
+            df_toff = pd.DataFrame(
+                file_npy_.T, columns=["timestamp", "toff1", "toff2", "n"]
+            )
+            timestamps_magic_min, timestamps_magic_max = (
+                float(min(timestamps_magic)) / 1e9,
+                float(max(timestamps_magic)) / 1e9,
+            )
+            timestamps_magic_min, timestamps_magic_max = (
+                timestamps_magic_min,
+                timestamps_magic_max,
+            )
+            df_toff = df_toff.query(
+                "@timestamps_magic_min<timestamp<@timestamps_magic_max"
+            )
+            t_plus = df_toff["timestamp"].values
+            time_offset_plus = df_toff["toff1"].values
+            n_coinc_plus = df_toff["n"].values
+            select = n_coinc_plus > 7
+            x = t_plus[select]
+            t0 = min(x)
+            x = x - t0
+            y = time_offset_plus[select]
+            popt, pcov = curve_fit(func1, x, y)
+            zero_offset = popt[0]  # sec
+            drift_speed = popt[1]  # sec/sec
+
+            df_magic["time_sec_sum"] = (
+                df_magic["time_sec"] + df_magic["time_nanosec"] * 1e-9 - t0
+            )
+            df_magic["time_shift"] = df_magic["time_sec_sum"] + (
+                zero_offset + df_magic["time_sec_sum"] * drift_speed
+            )
+            seconds = np.array([Decimal(str(time)) for time in df_magic["time_shift"]])
+
+            timestamps_magic = seconds * SEC2NSEC
+            # import matplotlib.pyplot as plt
+            # plt.scatter(x,y)
+            # plt.plot(np.linspace(min(x),max(x),100),(np.linspace(min(x),max(x),100)*drift_speed+zero_offset))
+            # plt.show()
+
+            t0_sec = [Decimal(str(time)) for time in [t0]]
+            t0_sec = np.array(t0_sec) * SEC2NSEC
+            t0_sec = u.Quantity(t0_sec, unit="ns", dtype=int)
+            timestamps_lst = timestamps_lst_arr[tel_id - 2] - t0_sec
```

Comment:
this is using tel_id = 2 and 3 for MAGIC-I and MAGIC-II, better to read it from the config